src/spec.c

```diff
@@ -2652,17 +2712,12 @@ int IndexSpec_UpdateDoc(IndexSpec *spec, RedisModuleCtx *ctx, RedisModuleString
   Document_Free(&doc);
 
   spec->stats.totalIndexTime += hires_clock_since_usec(&t0);
+
+  RedisSearchCtx_UnlockSpec(&sctx);
   return REDISMODULE_OK;
 }
 
-int IndexSpec_DeleteDoc(IndexSpec *spec, RedisModuleCtx *ctx, RedisModuleString *key) {
-
```

Comment:
since this was deleted, this function returns only `REDISMODULE_OK`

Reply:
the existing behavior was to return `REDISMODULE_ERR` if the doc is not existing